pywinauto/application.py

```diff
@@ -839,6 +840,15 @@ def __connect(self, **kwargs):
     def connect(self, **kwargs):
         """Connects to an already running process"""
         connected = False
+
+        if 'timeout' in kwargs:
+            try:
+                timeout = float(kwargs['timeout'])
+                del kwargs['timeout']
+                wait_until_passes(timeout, Timings.app_connect_retry, partial(self.connect, **kwargs), (ValueError, ProcessNotFoundError))
+            except Exception as e:
+                raise e

```

Comment:
It seems we don't need `try-except` block here because we do nothing with this error, just re-raising.